addons/point_of_sale/static/src/app/navbar/closing_popup/closing_popup.js

```diff
@@ -92,110 +109,112 @@ export class ClosePosPopup extends AbstractAwaitablePopup {
     async downloadSalesReport() {
         return this.report.download("point_of_sale.sale_details_report", [this.pos.pos_session.id]);
     }
-    handleInputChange(paymentId, amount) {
-        if (!this.env.utils.isValidFloat(amount)) {
-            return;
-        }
-        let expectedAmount;
-        if (paymentId === this.defaultCashDetails?.id) {
-            this.moneyDetails = null;
+    setManualCashInput(amount) {
+        if (this.env.utils.isValidFloat(amount) && this.moneyDetails) {
             this.state.notes = "";
-            expectedAmount = this.defaultCashDetails.amount;
-        } else {
-            expectedAmount = this.otherPaymentMethods.find((pm) => paymentId === pm.id).amount;
+            this.moneyDetails = null;
         }
-        this.state.payments[paymentId].counted = parseFloat(amount);
-        this.state.payments[paymentId].difference = this.env.utils.roundCurrency(
-            this.state.payments[paymentId].counted - expectedAmount
-        );
     }
-    hasDifference() {
-        return Object.entries(this.state.payments).find((pm) => pm[1].difference != 0);
+    getDifference(paymentId) {
+        const counted = this.state.payments[paymentId].counted.toString();
```

Comment:
counted should always a formatted number (type string). No need to call toString on it.